### DISCUSSION TITLE
Hard Drive Space Requirements?

### DISCUSSION THREAD

**Discussion by beef623** [PlummersSoftwareLLC/Primes#394] [Q&A]
How much hard drive space is needed to run the benchmarks? I stopped running them after the second and it was already taking about 2 Gb.

**Comment by marghidanu** [comment]
Keep in mind the benchmark is using Docker images, some images are larger than others. Although we strive to optimize images both in build time and size, not all the base images are small.

**Reply by beef623** [reply]
I reconfigured my Docker install to change where it stores its data and it's running again. It may be worth adding a note with an estimate in either the readme or the benchmark file though as the default location was in /var which is typically on a smaller partition unless the system is built for running Docker.

**Reply by marghidanu** [reply]
Maybe we can add a note in the FAQ about this or at least a requirement for disk size.

**Reply by rbergen** [reply]
The disk size requirement may be a tough one to keep up-to-date, as it can change with every language or even solution that gets added, particularly with the more niche/esoteric ones.

**Reply by beef623** [reply]
It's not really feasible to keep exact, but a warning would be nice. I know the default partitioning setup for several Linux distributions uses around a 40-50 Gb partition for the root which is where the default Docker storage location is. It was vastly larger than I expected it to be. Mine is sitting at ~32 Gb right now.

**Reply by marghidanu** [reply]
@rbergen I believe a warning should be enough.

**Comment by odmir** [comment]
On my machine the images on disk currently occupy ~20GB, but under preferences it reports a disk image size of ~34GB.
